### PR TITLE
make connection to minio configurable

### LINKED ISSUE #361: Support insecure minio connections
/kind feature
For local deployments (e.g. with Kubeflow), TLS is not configured for minio. When trying to deploy a model stored there, kfserving issues SSLError messages.

```Caused by SSLError(SSLError(1, '[SSL: WRONG_VERSION_NUMBER] wrong version number (_ssl.c:1056)')```

This is due to a hard-coded setting in [storage.py](https://github.com/kubeflow/kfserving/blob/master/python/kfserving/kfserving/storage.py#L194), with no way to override it.

**Describe the solution you'd like**
It should be possible to allow insecure connections to minio, having secure connection as default. The user should be able to do this either by specifying an S3_ENDPOINT with http scheme or by using a flag explicitly.

### PR DESCRIPTION
**Pull request by xaniasd**
**What this PR does / why we need it**:
Allows using insecure minio storage backend by specifying http as scheme in S3_ENDPOINT environment variable or explicitly by setting USE_SSL to False. Default setting remains secure.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #361 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
Allow use of insecure minio storage backend for local development/testing.
```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/362)

### CODE REVIEW

File: python/kfserving/kfserving/storage.py [hunk 1]
```diff
@@ -193,9 +193,10 @@ def _download_local(uri, out_dir=None):
     @staticmethod
     def _create_minio_client():
         # Remove possible http scheme for Minio
-        url = re.compile(r"https?://")
-        minioClient = Minio(url.sub("", os.getenv("S3_ENDPOINT", "")),
+        url = urlparse(os.getenv("S3_ENDPOINT", ""))
```
Comment (yuzisun) [inline python/kfserving/kfserving/storage.py]: `S3_ENDPOINT` does not contain the scheme, it is actually `AWS_ENDPOINT_URL`(https://github.com/kubeflow/kfserving/blob/master/pkg/controller/kfservice/resources/credentials/s3/s3_secret.go#L118)

File: python/kfserving/kfserving/storage.py [hunk 2]
```diff
@@ -193,9 +193,10 @@ def _download_local(uri, out_dir=None):
...
+        use_ssl = url.scheme=='https' if url.scheme else bool(os.getenv("USE_SSL", True))
```
Comment (yuzisun) [inline python/kfserving/kfserving/storage.py]: We have env variable `S3_USE_HTTPS`(https://github.com/kubeflow/kfserving/blob/master/pkg/controller/kfservice/resources/credentials/s3/s3_secret.go#L32)

File: python/kfserving/kfserving/storage.py (hunk 1 above)
Comment (xaniasd) [reply python/kfserving/kfserving/storage.py]: thanks! fixed

File: python/kfserving/kfserving/storage.py (hunk 2 above)
Comment (xaniasd) [reply python/kfserving/kfserving/storage.py]: fixed!

File: python/kfserving/kfserving/storage.py [hunk 3]
```diff
@@ -193,9 +193,10 @@ def _download_local(uri, out_dir=None):
     @staticmethod
     def _create_minio_client():
         # Remove possible http scheme for Minio
-        url = re.compile(r"https?://")
-        minioClient = Minio(url.sub("", os.getenv("S3_ENDPOINT", "")),
+        url = urlparse(os.getenv("AWS_ENDPOINT_URL", ""))
```
Comment (yuzisun) [inline python/kfserving/kfserving/storage.py]: looks `urlparse` is not imported according to lint report.

Comment (xaniasd) [reply python/kfserving/kfserving/storage.py]: hi @yuzisun, sorry about that, I was betrayed by my IDE :) anyway, this is fixed now but I still get errors that seem unrelated?

**Comment by yuzisun** [comment]
Thanks for the contribution @xaniasd ! Looks great!

/lgtm
/approve

**Pull request merged**